addon/v1alpha1/types_clustermanageraddon.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Cluster"
+
+// ClusterManagerAddOn represents the registration of an addon
+// It allows the user to discover which addon is available for the cluster manager.
```

Comment:
...asking because I think this type needs to be contextualized with other types in the API here. It's not clear exactly what "the cluster manager" is specifically in this context.